server/sonar-server/src/main/java/org/sonar/server/plugins/ServerPluginRepository.java

```diff
@@ -124,9 +126,15 @@ public void stop() {
     loader.unload(pluginInstancesByKeys.values());
     pluginInstancesByKeys.clear();
     pluginInfosByKeys.clear();
+    keysByClassLoader.clear();
     started.set(true);
   }
 
+  @CheckForNull
+  public String getPluginKey(ClassLoader cl) {
```

Comment:
I don't like having a method that needs a ClassLoader. It should be best to hide this.